src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,239 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { deleteTodo, updateTodoStatus, updateTodoTitle } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+import cn from 'classnames';
+
+type Props = {
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+  filteredTodos: Todo[];
+  todos: Todo[];
+  tempTodo: Todo | null;
+  setLoadingTodosId: React.Dispatch<React.SetStateAction<number[]>>;
+  loadingTodosId: number[];
+  setEditingTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+  editingTodo: Todo | null;
+};
+
+export const TodoList: React.FC<Props> = ({
+  setTodos,
+  setErrorMessage,
+  filteredTodos,
+  todos,
+  tempTodo,
+  setLoadingTodosId,
+  loadingTodosId,
+  setEditingTodo,
+  editingTodo,
+}) => {
+  const [query, setQuery] = useState('');
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const cancelEditing = useCallback(() => {
+    setQuery('');
+    setLoadingTodosId([]);
+    setEditingTodo(null);
+  }, [setLoadingTodosId]);
+
+  useEffect(() => {
+    const handleKeyUp = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        event.preventDefault();
+        cancelEditing();
+      }
+    };
+
+    document.addEventListener('keyup', handleKeyUp);
+
+    return () => {
+      document.removeEventListener('keyup', handleKeyUp);
+    };
+  }, [cancelEditing]);
+
+  const handleOnClickDelete = async (todoForDelate: Todo) => {
+    setLoadingTodosId(prev => [...prev, todoForDelate.id]);
+    try {
+      await deleteTodo(todoForDelate.id);
+      setTodos(prevTodos =>
+        prevTodos.filter(todo => todoForDelate.id !== todo.id),
+      );
+    } catch (error) {
+      setErrorMessage('Unable to delete a todo');
+    } finally {
+      setLoadingTodosId([]);
+    }
+  };
```

Comment:
why do you need to pass here setLoadingTodosId, setTodos, todos and etc. to this component, if you can create this handleOnClickDelete function in the App.tsx and just pass this function as a prop

Reply:
try to fix it everywhere